File: UCF HSPT Project/src/y2009/smart.java
// Arup Guha
// 5/4/09
// Solution to 2009 UCF HS Programming Contest Problem: Smart.

import java.util.*;
import java.io.*;

public class smart
{
	public static void main(String[] args) throws Exception
	{
		Scanner in;
		
		// Open the input file. If it's not there, read from the keyboard.
		try
		{
			in = new Scanner(new File("smart.in"));
		}
		catch(FileNotFoundException e)
		{
			in = new Scanner(System.in);
		}
		
		// Read in the number of data sets.
		int sets = in.nextInt();
		
		// Read through each set and output the appropriate message.
		for(int s = 1; s <= sets; s++)
			if(in.nextInt() == in.nextInt())
				System.out.println("The Old Knife-Hits-the-Target Trick");
			else
				System.out.println("Missed it by THAT much");
	
	
	}
}

File: UCF HSPT Documents/2009/Solutions/smart.cpp
/**
 * 2009 UCF High School Programming Tournament
 *
 * C++ solution to "Obtain Intelligence"
 *
 */
#include <fstream>
#include <iostream>

int main()
{
   // Set up input and output for this program.
   //
   // The input will be from a file called smart.in; we'll call this 'fin' for
   // "[f]ile [in]put."
   //
   // The output will be to the console, also called "standard output" and
   // provided by C++ as the 'cout' object (meaning "[c]onsole [out]put").
   // Note that the 'ifstream' class and the 'cout' object are both in the
   // 'std' ("standard") namespace, so we have to put "std::" in front of
   // them.
   std::ifstream fin( "smart.in" );

   // The problem defines positive integer 'n' as the number of throws
   unsigned int n;

   // Read in the number of throws from the input file.
   fin >> n;

   // Use a loop, counting from 0 to n-1, to process each throw
   for( unsigned int i = 0; i < n; ++i )
   {
      // For each throw, there will be two positive integers 't' and 'p'.
      unsigned int t;  // 't' is the position of the target
      unsigned int p;  // 'p' is the position of Agent 13's throw

      // Read in 't' and 'p' from the input file.
      // C++ will automatically take care of the spaces and line breaks.
      fin >> t >> p;

      // Check whether the target was hit.
      if( t == p )
      {
         // It's a hit! Output the correct message.
         // Be careful with capitalization, punctuation, and spacing!
         std::cout << "The Old Knife-Hits-the-Target Trick" << std::endl;
      }
      else
      {
         // It's a miss. Output the correct message.
         // Be careful with capitalization, punctuation, and spacing!
         std::cout << "Missed it by THAT much" << std::endl;
      }
   }
}

